### ISSUE TITLE
Add Prometheus metrics support to the decoder

### ISSUE THREAD

**Issue by dvdplm** [0xPolygonZero/proof-protocol-decoder#23]
In the wake of plonky2 Issue #1513, add metrics support to the decoder as well. 

- [ ] Investigate an initial set of metrics we want to track
- [ ] Add prometheus support, including configuration and sane defaults

**Comment by BGluth** [comment]
Yeah Prometheus metrics would be very nice to have (and probably low-effort to setup).
